chat2_client.lua

```diff
@@ -38,18 +36,12 @@ function clear()
 end
 
 function show(bool)
-  if chatInstanceLoaded ~= true then
-    if chatInstanceLoading ~= true then
-      create()
-      setTimer(show, 300, 1, bool)
-    else
-      setTimer(show, 300, 1, bool)
-    end
-  end
-
-  local eval = "show(" .. tostring(bool) .. ");"
-  execute(eval)
-  setElementData(localPlayer, "chat2IsVisible", bool)
+    if chatInstanceLoaded then return end
+    if not chatInstanceLoading then create() end
+    setTimer(show, 300, 1, bool)
+    local eval = "show(" .. tostring(bool) .. ");"
+    execute(eval)
+    setElementData(localPlayer, "chat2IsVisible", bool)
```

Comment:
Show function does 3 steps. It loads CEF if it doesn't, then it waits until CEF load and only then it shows/hides a chat. If you call show(true / false) second time after CEF load, it will skip first 2 steps.

Current changes didn't work because it didn't wait for CEF load but executes show() and then ```if chatInstanceLoaded then return end``` blocks another execution of show that was passed to setTimer as a callback. So chatInstance was loaded, but chat will be not shown.

Try to test your changes with other resource, that should be loaded into mta after chat2 resource and should execute next code:
```
-- clientside
function handler()
    exports.chat2:output("hello world")
end

addEventHandler("onClientResourceStart", resourceRoot, handler)
```